cranelift-codegen/meta/src/isa/x86/legalize.rs

```diff
@@ -290,4 +301,84 @@ pub fn define(shared: &mut SharedDefinitions, x86_instructions: &InstructionGrou
     );
 
     group.build_and_add_to(&mut shared.transform_groups);
+
+    let mut narrow = TransformGroupBuilder::new(
+        "x86_narrow",
+        r#"
+    Legalize instructions by narrowing.
+
+    Use x86-specific instructions if needed."#,
+    )
+    .isa("x86")
+    .chain_with(shared.transform_groups.by_name("narrow").id);
+
+    // SIMD
+    let broadcast_immediate_u8 = Literal::constant(uimm8, 0x00);
+    let broadcast_immediate_f64 = Literal::constant(ieee64, 0x00);
+    let lane_one = Literal::constant(uimm8, 0x01);
+    let b = var("b");
+    let c = var("c");
+    let d = var("d");
+
+    // SIMD splat: 8-bits
+    for ty in LaneTypeIterator::new().filter(|t| t.lane_bits() == 8) {
+        let splat_x8x16 = splat.bind_vector(ty, 128 / ty.lane_bits());
```

Comment:
`lane_bits` is always 8 here right?

Reply:
In my next set of commits for `extractlane` I have a commit that changes `bind_vector` so that only the lane type is passed; insided `bind_vector` the number of lanes is calculated based on 128-bit registers. The way I see it, if we have to have different vector sizes later we can change that but for now having the calculation for the number of lanes in one place is better than many places and also better than manually setting the number of lanes at each call site--less opportunity for mistakes. Why don't we fix this in the next PR?